### ISSUE TITLE
Bilibili - Channels don’t load

### ISSUE THREAD

**Issue by matsumotokenji** [futo-org/grayjay-android#1273]
### What happened?

All channels main page broken after an plugin update
![image](https://github.com/user-attachments/assets/29312a51-c044-4a7e-a05a-9ff7c9ca8e9a)

### Grayjay Version

253

### What plugins are you seeing the problem on?

BiliBili (CN)

### Plugin Version

14

### When do you experience the issue?

- [X] While logged in
- [X] While logged out
- [ ] N/A

**Comment by codebymatter** [comment]
Use the Reinstall button to rollback the plugin version to the last version. I do not use BiliBili so cannot comment on whether I have the same issue. ****

**Comment by matsumotokenji** [comment]
reinstalled and rollbacked to version 13 but same issue, seems like issue due to changes on bilibili site

**Comment by codebymatter** [comment]
I tried Bilibili and played a random video and it seems to be working just fine. I am in N. America, not using a VPN.

![Screenshot_20240830-010729_Grayjay](https://github.com/user-attachments/assets/0ea266b9-8a98-45ff-8b47-275d24d66072)
![Screenshot_20240830-010845_Grayjay](https://github.com/user-attachments/assets/c6a006aa-beb0-4448-b3e3-4338329d5dfd)

**Comment by matsumotokenji** [comment]
seems back to normal now

**Comment by codebymatter** [comment]
I did:

![Screenshot_20240830-140836_Grayjay](https://github.com/user-attachments/assets/9001d5ae-1e3a-4d3c-9fcb-f834a6a3d4d0)
![Screenshot_20240830-141006_Grayjay](https://github.com/user-attachments/assets/b4f87210-287f-409f-9762-f5832258f829)
![Screenshot_20240830-141045_Grayjay](https://github.com/user-attachments/assets/c2dba187-62b8-45af-bf90-d22da637a7c3)

**Comment by matsumotokenji** [comment]
> I did:
> 
> ![Screenshot_20240830-140836_Grayjay](https://private-user-images.githubusercontent.com/59785399/363287140-9001d5ae-1e3a-4d3c-9fcb-f834a6a3d4d0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzMzY4MjIsIm5iZiI6MTcyNTMzNjUyMiwicGF0aCI6Ii81OTc4NTM5OS8zNjMyODcxNDAtOTAwMWQ1YWUtMWUzYS00ZDNjLTlmY2ItZjgzNGE2YTNkNGQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAzVDA0MDg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0OGEyYTJlY2QwNWY5MTRkMmU5YjdiMmE2NDlkNzU1Mjk3MWI3NzlmYmIwYjQ1YzhmNjA5NGEzMzQyYjRjM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EgcGW92sZQCPSfywmgaTcfeYOTL8W-9q97-B_rKxMiY) ![Screenshot_20240830-141006_Grayjay](https://private-user-images.githubusercontent.com/59785399/363287226-b4f87210-287f-409f-9762-f5832258f829.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzMzY4MjIsIm5iZiI6MTcyNTMzNjUyMiwicGF0aCI6Ii81OTc4NTM5OS8zNjMyODcyMjYtYjRmODcyMTAtMjg3Zi00MDlmLTk3NjItZjU4MzIyNThmODI5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAzVDA0MDg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwNzRiZWJmNjc3ODQ4ZWM3Y2I1N2IwNTJiNGE4MjU2NWZhNTk4Y2U0ODgwZDMxMDdkZjEyM2Q5YzgzOTA5NTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ahjHKaRc8tXkYBx3hBvkB7CWyED-g4ATh5-6AEtmS88) ![Screenshot_20240830-141045_Grayjay](https://private-user-images.githubusercontent.com/59785399/363287282-c2dba187-62b8-45af-bf90-d22da637a7c3.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzMzY4MjIsIm5iZiI6MTcyNTMzNjUyMiwicGF0aCI6Ii81OTc4NTM5OS8zNjMyODcyODItYzJkYmExODctNjJiOC00NWFmLWJmOTAtZDIyZGE2MzdhN2MzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAzVDA0MDg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmZTc5ZjI3Y2RiZGMwYjViZGU0YThhZDUyZGUxNDJkYWE1MmQyYTNjMDViMjc4ZTkzZjgxODFlNzRhMWI4YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x7bBaMo8in0lUPM9sMJEdLJdXUMEhTMHfmbYWGqVeK8)

welp, it worked fine for a few days and stop working. not using vpn btw i can visit bilibili normally via official app and web.
![Screenshot_20240903_120753](https://github.com/user-attachments/assets/710944d1-619c-4712-8dc2-f4868a11be96)

**Comment by kaidelorenzo** [comment]
@matsumotokenji do you still have this issue? I'm trying to replicate it but can't. My guess is that it is some kind of rate limiting. Signing into an account should fix it if that's an option for you.

**Comment by matsumotokenji** [comment]
> @matsumotokenji do you still have this issue? I'm trying to replicate it but can't. My guess is that it is some kind of rate limiting. Signing into an account should fix it if that's an option for you.

yes, this is not 24hr problem, sometimes cant work on mornrng but fine at evening, sometimes other way around. seems like bilibili may have different themes or api on different time period

and btw casting from bilibili is broken to fcast.

**Comment by Zvonimir-FUTO** [comment]
Tried opening this channel for whole week from time to time but I couldn't have replicate it not even once.

**Comment by matsumotokenji** [comment]
anyway i can help to provide debug information?

**Comment by matsumotokenji** [comment]
thanks, it fails to load regardless i am logged off or on. Maybe this is ip or region specific.

**Comment by kaidelorenzo** [comment]
i might try to replicate with via a VPN. what general region are you?

**Comment by matsumotokenji** [comment]
malaysia happens on at least 2 isp, sometimes some channels worked fine and some not, usually fails to work on morning.

like now
https://www.bilibili.com/video/BV1f2mpYdEte
![Screenshot_20241112_114047](https://github.com/user-attachments/assets/58dcf02f-58a8-48d1-98f6-c554e5ccf730)
